base/siteorigin-widget.class.php

```diff
@@ -727,23 +727,33 @@ private function save_css( $instance ){
 
 		$css = $this->get_instance_css($instance);
 
-		if( !empty($css) ) {
-
+		if ( ! empty( $css ) ) {
 			if ( WP_Filesystem() ) {
 				global $wp_filesystem;
 				$upload_dir = wp_upload_dir();
 
 				if ( ! $wp_filesystem->is_dir( $upload_dir['basedir'] . '/siteorigin-widgets/' ) ) {
-					$wp_filesystem->mkdir( $upload_dir['basedir'] . '/siteorigin-widgets/' );
+					$directory_created = $wp_filesystem->mkdir( $upload_dir['basedir'] . '/siteorigin-widgets/' );
+					if ( ! $directory_created ) {
+						$add_cache = true;
+					}
```

Comment:
This could be simplified to avoid all the conditionals here and below. Just assign the result of the various file write attempts to the `$add_cache` variable and check the result at the end. I.e.
`$add_cache = ! $wp_filesystem->mkdir( $upload_dir['basedir'] . '/siteorigin-widgets/' );`
and
`$add_cache = ! $wp_filesystem->put_contents(...`

Reply:
Oh whoops. For some reason, I had thought this was causing errors to be logged. I just double checked the initial thread and I was mistaken.